src/app/routes/fetchPageData/index.js

```diff
@@ -25,63 +22,50 @@ export const getUrl = pathname => {
   if (!pathname) return '';
 
   const params = isLive() ? '' : getQueryString(pathname);
-
   const basePath = getUrlPath(pathname);
 
-  return `${baseUrl}${basePath.replace(ampRegex, '')}.json${params}`;
+  return `${baseUrl}${basePath.replace(ampRegex, '')}.json${params}`; // Remove .amp at the end of pathnames for AMP pages.
 };
 
-const handleResponse = async response => {
+const handleResponse = url => async response => {
   const { status } = response;
 
-  if (status === STATUS_CODE_OK) {
-    const pageData = await response.json();
-    const type = path(['metadata', 'type'], pageData);
-    const processedPageData = await preprocess(
-      pageData,
-      getPreprocessorRules(type),
-    );
+  if (upstreamStatusCodesToPropagate.includes(status)) {
+    if (status === STATUS_NOT_FOUND) {
+      logger.error(`Data not found when requesting ${url}`);
+    }
 
     return {
       status,
-      pageData: processedPageData,
+      ...(status === STATUS_OK && {
+        json: await response.json(),
+      }),
     };
   }
 
-  return { status };
-};
-
-const checkForError = pathname => ({ status, pageData }) => {
-  const isHandledStatus = upstreamStatusCodesToPropagate.includes(status);
-
-  if (isHandledStatus) {
-    return { status, pageData };
-  }
-  logger.error(
-    `Unexpected upstream response (HTTP status code ${status}) when requesting ${pathname}`,
+  throw new Error(
+    `Unexpected upstream response (HTTP status code ${status}) when requesting ${url}`,
   );
-  throw new Error();
 };
 
-const handleError = error => {
-  if (error.message) {
-    logger.error(error.message);
-  }
+const handleError = e => {
+  const error = e.toString();
 
-  const status = onClient()
-    ? STATUS_CODE_BAD_GATEWAY
-    : STATUS_CODE_INTERNAL_SERVER_ERROR;
+  logger.error(error);
 
-  return { error, status };
+  return {
+    error,
+    status: onClient() ? STATUS_BAD_GATEWAY : STATUS_INTERNAL_SERVER_ERROR,
+  };
 };
 
 const fetchData = pathname => {
-  const url = getUrl(pathname); // Remove .amp at the end of pathnames for AMP pages.
+  const url = getUrl(pathname);
+
   logger.info(`DataRequest: [${url}]`);
 
   return fetch(url)
-    .then(handleResponse)
-    .then(checkForError(url))
+    .then(handleResponse(url))
```

Comment:
With the advent of `async` do we need to use promises here? We could change this to:
```javascript
  try {
    const response = await fetch(url);
    return handleReponse(url, response)
  } catch (error) {
    return handleError(error)
  }
```

Reply:
Following the mob review I'd just like to add an argument for using async/await exclusively over promises, specifically referring to error handling. Given the following script:

```javascript
const func1 = async throwSomething => {
  if (throwSomething) {
    throw new Error('eek');
  }
};

func1().then(() => {
  console.log('All good');
});

func1(true).then(() => {
  console.log('All good');
});
```

You get this output:
```
All good
(node:72672) UnhandledPromiseRejectionWarning: Error: eek
    at func1 (/Users/bennea27/Documents/workspace/simorgh/errorExample.js:3:11)
    at Object.<anonymous> (/Users/bennea27/Documents/workspace/simorgh/errorExample.js:11:1)
    at Module._compile (internal/modules/cjs/loader.js:955:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:991:10)
    at Module.load (internal/modules/cjs/loader.js:811:32)
    at Function.Module._load (internal/modules/cjs/loader.js:723:14)
    at Function.Module.runMain (internal/modules/cjs/loader.js:1043:10)
    at internal/main/run_main_module.js:17:11
(node:72672) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
(node:72672) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
- The first call throws no error, so `all good` is output to the console
- The second call throws an error and the `all good` continuation is not called, that is to be expected.
  - The crucial bit though is that error is not caught or handled anywhere, it merely outputs a warning to console. 

So the important part is that _every time_ you use a promise you must remember to use catch to handle any errors (as you have in your case). If you use a await, even if you don't use a try catch block, the exception will be thrown up the call stack and in a react app can be handled by an error boundary or suitable higher level exception handling.

I hope that makes sense? I wanted to give some justification to my 'promises are deprecated' grand statement I made 😈